src/aggregate/aggregate_exec.c

```diff
@@ -161,8 +161,18 @@ static int sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
   RedisModule_ReplyWithArray(outctx, REDISMODULE_POSTPONED_ARRAY_LEN);
 
   rc = rp->Next(rp, &r);
-  RedisModule_ReplyWithLongLong(outctx, req->qiter.totalResults);
+  if (rc == RS_RESULT_TIMEDOUT) {
+    if (!(req->reqflags & QEXEC_F_IS_CURSOR)) {
+      RedisModule_ReplyWithSimpleString(outctx, "Timeout limit was reached");
+    } else {
+      rc = RS_RESULT_OK;
+      RedisModule_ReplyWithLongLong(outctx, req->qiter.totalResults);
```

Comment:
@ashtul so I suspect now that the cursor might not return the exact number that you ask if timeout reached, and with cursor you can continue from the same place you stopped right? We need to document it, kind of behavior change.
@K-Jo FYI